### PR TITLE
Create ROS reference page

### LINKED ISSUE #29: Create docs page Reference/ROS
### Purpose
How to install, set up, and ROS for our project

### Page Tree
```yml
nav:
  ...
  - Reference:
    ...
    - ROS: reference/ros.md
```

### Contents
- Point or add to installation instructions, relevant readings, relevant tutorials in ROS documentation

### Relevant Resources
- https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools.html
- https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries.html
- https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Launch-Main.html
- https://github.com/UBCSailbot/local-pathfinding/wiki/How-To-Install
- https://answers.ros.org/question/286880/how-to-make-roscd-go-to-catkin_wsdevel-instead-of-optroskinetic/

### PR DESCRIPTION
**Pull request by patrick-5546**
* Resolves #29 
* Developed in conjunction with https://github.com/UBCSailbot/sailbot_workspace/pull/25

### CODE REVIEW

**Comment by patrick-5546** [comment]
@DFriend01 @hhenry01 

I have made some good progress on our ROS documentation. If you have time, can you try going through the tutorials I listed, and let me know what you think? Much appreciated

#### Notes

- Use the `support_tutorials` branch of sailbot_workspace
- So far I have only verified that tutorials up to and including "Creating a workspace" work

Side note: Going through these first few tutorials, I learned so much about ROS that I didn't know before. I'm already thinking of new ways we can use ROS to make our new project even better :)

File: docs/reference/ros.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+multiple machines, tools for testing and visualization, and much more [^2].
+
+We uses ROS because it is open-source, language-agnostic, and built with cross-collaboration in mind.
+It enables our subteams to work independently on well-defined components of our software system,
+without having to worry about the hardware it runs on or the implementation of other components.
+
+[The official ROS 2 documentation](https://docs.ros.org/en/eloquent/index.html) contains everything you need
+to get started using ROS. From it we have hand-picked the resources that are most relevant to our current and expected
+future usage of ROS assuming that you use [our preconfigured workspace](https://github.com/UBCSailbot/sailbot_workspace).
+Installing and setting up ROS without our workspace is outside the scope of this site.
```
Comment (DFriend01) [inline docs/reference/ros.md]: Consider adding a quick blurb saying that installation instructions are on the ROS official documentation if they are interested in running it on bare metal.

File: docs/reference/ros.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+To get our workspace configuration running on your computer:
+
+1. Set it up by following the instructions in [its README](https://github.com/UBCSailbot/sailbot_workspace#setup)
+2. Uncomment the ROS 2 tutorials section in [`.devcontainer/Dockerfile`](https://github.com/UBCSailbot/sailbot_workspace/blob/main/.devcontainer/Dockerfile),
+   then run the "Remote-Containers: Rebuild Container" VS Code command, to install the tutorials' dependencies
+3. Uncomment the ROS 2 tutorials section in [`src/new_project.repos`](https://github.com/UBCSailbot/sailbot_workspace/blob/main/src/new_project.repos),
+   then run the "setup" VS Code task, to clone the repositories used in the tutorials
+
+Our workspace configuration contains easier methods of accomplishing some of the tutorial steps, or eliminates the need
+for them altogether.
```
Comment (DFriend01) [inline docs/reference/ros.md:27]: Any examples of what functionalities are accomplished using the workspace? If there are not too many, a quick exhaustive list would be nice.

**Review by DFriend01 — CHANGES_REQUESTED** [review]
I only had one comment. Once that is addressed, this PR looks good to me.

**Review by hhenry01 — APPROVED** [review]
Looks good to me.

**Review by DFriend01 — APPROVED** [review]

**Pull request merged**